### DISCUSSION TITLE
消息按组分类获取消息信息

### DISCUSSION THREAD

**Discussion by mhdy2233** [tangyoha/telegram_media_downloader#141] [Ideas]
能否添加一个功能将每组消息或者说一条消息内包含的多个图片、视频、文件等分到一个文件夹里，并且获取这条消息下的文字以及他的id和时间
Can you add a function to divide each group of messages or multiple pictures, videos, files, etc. contained in a message into a folder, and obtain the text under this message as well as its id and time?

**Comment by tangyoha** [comment]
@mhdy2233 hi，功能请求请提issue

**Comment by tangyoha** [comment]
你好！关于将消息组（media group）中的多个文件分到一个文件夹并获取消息文本的功能，这是一个很好的建议。

Hello! Regarding grouping multiple files from a media group into one folder and capturing message text, this is a great feature request.

**当前的部分支持**：

目前可以通过 `file_path_prefix` 配置来组织文件结构：

```yaml
file_path_prefix:
  - chat_title
  - media_datetime  # 按日期分组
  - message_id      # 按消息ID分组
```

**按消息ID分组**：

如果使用 `message_id` 作为路径前缀，同一条消息的多个文件会自动放在同一个文件夹中：

```yaml
file_path_prefix:
  - chat_title
  - message_id
```

这样会生成类似的结构：
```
downloads/
  ChatName/
    12345/  # 消息ID
      photo1.jpg
      photo2.jpg
      video.mp4
```

**关于保存消息文本和元数据**：

目前程序不会自动保存消息的文本内容。建议作者可以添加类似的配置：

```yaml
save_message_metadata: true  # 保存消息元数据
metadata_format: json        # 或 txt
```

这样会在每个消息文件夹中生成一个 `metadata.json`：
```json
{
  "message_id": 12345,
  "date": "2025-03-04T10:30:00",
  "text": "消息文本内容",
  "caption": "图片说明",
  "media_count": 3
}
```

这是一个非常实用的功能，建议提交 Feature Request！
